### ISSUE TITLE
dependencies check of php version doesn’t work correct 

### ISSUE THREAD

**Issue by GitStefanN** [owncloud/core#14392]
I tried to activate the gallery+ app, but I get the Message:
"This app cannot be installed because the following dependencies are not fulfilled: PHP with a version lower than 5.5 is required.”
The gallery+ app does support php 5.5. (see https://github.com/interfasys/galleryplus/blob/master/appinfo/info.xml)
According to @oparoz there is maybe a bug in ownCloud's check routine.
The configuration of my server: Ubuntu 14.04 LTS, Apache/2.4.7 (Ubuntu), PHP 5.5.9-1ubuntu4.6, ownCloud 8.0 (stable)

**Comment by oparoz** [comment]
[ownCloud documentation](http://doc.owncloud.org/server/8.0/developer_manual/app/info.html)
```
<dependencies>
        <php min-version="5.4" max-version="5.5"/>
...
        <owncloud min-version="6.0.4" max-version="8"/>
    </dependencies>
```
[Gallery+'s info.xml](https://github.com/interfasys/galleryplus/blob/master/appinfo/info.xml)
```
<dependencies>
	<php min-version="5.4" max-version="5.5"/>
	<owncloud min-version="8"/>
</dependencies>
```

**Comment by oparoz** [comment]
@DeepDiver1975 
https://github.com/owncloud/core/blob/disable-non-shipped-apps-master/lib/private/app/dependencyanalyzer.php#L63
```
if (version_compare($this->platform->getPhpVersion(), $maxVersion, '>')) {
```
translates to 
if PHP 5.5.9 not greater than PHP 5.5, then fail, but PHP 5.5 is allowed as the max version.

**Comment by Raydiation** [comment]
This is actually not a bug and works as expected, since 5.1 > 5, so 5 < 5.1

I dont know why you set 5.5 as max version, you should remove that .

@DeepDiver1975 As for the usecase: what we actually might want to do is to strip both numbers to the same length, e.g. min version 5, current version 5.1.2 should strip the second version to 5, min version 5.1 and current version 5.1.1 should strip the current version to 5.1. In that case we want to use >= and <= so this becomes more intuitive.

**Comment by oparoz** [comment]
@Raydiation - It is a bug. If I say that the max supported version is 5.5, then the app check should pass on 5.5. That is a supported version.
If you want app devs to set 5.6 as the upper limit, you can rename the parameter to version-below-which-the-app-will-run

**Comment by DeepDiver1975** [comment]
fixed with #14529 -> close
